### PR TITLE
[SPARK-50324][PYTHON][CONNECT] Make `createDataFrame` trigger `Config` RPC at most once

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Get all configs in batch

### Why are the changes needed?
there are too many related configs in `createDataFrame`, they are fetched one by one (or group by group) in different branches:
1, it is possible no Config RPC is triggered, e.g. in this branch:
https://github.com/apache/spark/blob/26330355836f5b2dad9b7bd4c72d9830c7ce6788/python/pyspark/sql/connect/session.py#L502-L509

2, multiple Config RPCs for different configs, e.g. in this branch:
https://github.com/apache/spark/blob/26330355836f5b2dad9b7bd4c72d9830c7ce6788/python/pyspark/sql/connect/session.py#L599-L601

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-meng** [comment]
LGTM thank you!

File: python/pyspark/sql/connect/session.py [hunk 1]
```diff
@@ -706,9 +724,9 @@ def createDataFrame(
         else:
             local_relation = LocalRelation(_table)
 
-        cache_threshold = self._client.get_configs("spark.sql.session.localRelationCacheThreshold")
+        cache_threshold = conf_getter["spark.sql.session.localRelationCacheThreshold"]
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/session.py:727]: Shall we just get all the confs in batch eagerly? Seems like we should get the conf once anyway.

Comment (zhengruifeng) [reply python/pyspark/sql/connect/session.py:727]: there are 2 cases that we don't need the configs:
1, the local data is empty, and the schema is specified, it returns a valid empty df;
2, the creation fails due to some assertions

**Review by HyukjinKwon — COMMENTED** [review]
I think another way of doing this is to maintain a sized dictionary in Python side, and cache the value retrieved within the Python side.

e.g., 

- spark.get("a")
  - look up cacehd["a"] = v
    - if not, spark.get("a")

- spark.set("a", aa")
  - empty cache cached["a"]
  - spark.set("a")

and create a dictioanry with TTL and max size

**Comment by zhengruifeng** [comment]
It seems we don't need this helper class to achieve the goal, will have another try

**Comment by zhengruifeng** [comment]
we can get all configs before `if isinstance(data, pd.DataFrame)`

```
        (... , ...) = self._client.get_configs(
            "spark.sql.timestampType",
            "spark.sql.session.timeZone",
            "spark.sql.session.localRelationCacheThreshold",
            "spark.sql.execution.pandas.convertToArrowArraySafely",
            "spark.sql.execution.pandas.inferPandasDictAsMap",
            "spark.sql.pyspark.inferNestedDictAsStruct.enabled",
            "spark.sql.pyspark.legacy.inferArrayTypeFromFirstElement.enabled",
            "spark.sql.pyspark.legacy.inferMapTypeFromFirstPair.enabled",
        )
```

but after comparison, I think existing PR is more convenient to handle multiple configs.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by zhengruifeng** [comment]
thanks, merged to master

**Pull request closed without merging**